### PR TITLE
Workflow documentation, creating pull request

### LINKED ISSUE #2383: Refer to basic Github Fork+PR workflow in the docs
**Is your feature request related to a problem? Please describe.**

Every now and then, we can see that people who are new to `git`, GitHub, and open-source are facing difficulties with sending and updating a PRs while maintaining their local environment clean.

**Describe the solution you'd like**

I suggest adding a `note` \ `tip` \ `important` or a small section in the relevant locations of two of our contribution guides

- [Contributing Code to Cutter](https://cutter.re/docs/contributing/code/getting-started.html)
- [Contributing Documentation to Cutter](https://cutter.re/docs/contributing/docs/getting-started.html)

An example to such paragraph can look something like this:

"Contributing to Cutter require you to have a basic knowledge of git and the Github Fork & Pull-Requst workflow. If you aren't familiar with the subjects, we highly recommend to read the "[GitHub - Contributing to a Project](https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project)" on the official and detailed [Git Book](https://git-scm.com/book/en/v2)."

I suggest to expand this example a little bit in the "Contributing Code to Cutter" page, and shortening it a bit in the "Contributing Documentation to Cutter" page

For the "Contributing Documentation to Cutter" it is also good to mention and link to how to edit files using the Edit (pencil icon) button. A useful link can be found on [GitHub Documentation - Editing files](https://docs.github.com/en/enterprise/2.14/user/articles/editing-files-in-your-repository).

**Additional context**

If you think that there are better tutorials, please suggest them :)

### PR DESCRIPTION
**Pull request by 7h3qu1rkyb1t**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [ ] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**
since it is for both contributing to docs and contribute to code i thought it would be better to put it in common file contributing.rst

**Test plan (required)**

**Closing issues**
"closes #2383"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,11 +16,12 @@ Cutter is a free and open-source reverse engineering framework powered by [radar
...
 You can download the latest release [here](https://github.com/radareorg/cutter/releases).
- * macOS: Download the latest `.dmg` file or use [Homebrew Cask](https://github.com/Homebrew/homebrew-cask) `brew cask install cutter`.
- * Windows: Download the latest Zip archive.
- * Linux: Download the latest AppImage file. Then just make it executable and run it:
-   * `chmod +x <appimage_file>`
-   * `./<appimage_file>`
+
+* macOS: Download the latest `.dmg` file or use [Homebrew Cask](https://github.com/Homebrew/homebrew-cask) `brew cask install cutter`
+* Windows: Download the latest Zip archive.
+* Linux: Download the latest AppImage file. Then just make it executable and run it:
```
Comment (7h3qu1rkyb1t) [inline README.md:22]: nothing has changed here. may be my vim markdown extension did some auto formatting here

Comment (karliss) [reply README.md:22]: Please revert changes to this part. It is supposed to be nested list instead of single flat list.

Comment (7h3qu1rkyb1t) [reply README.md:22]: it is nested.

Comment (7h3qu1rkyb1t) [reply README.md:22]: may be comment makes it to look like they are in same line

Comment (7h3qu1rkyb1t) [reply README.md]: which editor do you guys use for markdown editing

File: README.md [hunk 2]
```diff
@@ -35,11 +35,17 @@ To deploy *cutter* using a pre-built `Dockerfile`, it's possible to use the [pro
...
 Cutter supports both Python and Native C++ plugins. Want to extend Cutter with Plugins? Read the [Plugins](https://cutter.re/docs/plugins) section on our documentation.
 
 ### Official and Community Plugins
+
 Our community built many plugins and useful scripts for Cutter such as the native integration of Ghidra decompiler or the plugin to visualize DynamoRIO code coverage. You can find more plugins in the [following list](https://github.com/radareorg/cutter-plugins). Don't hesitate to extend it with your own plugins and scripts for Cutter.
 
+### Contribute
+
+Take a look at [contributing docs](https://cutter.re/docs/contributing.html).
+
```
Comment (ITAYC0HEN) [inline README.md:48]:
```suggestion
### Contributing to Cutter
Our [Contributing docs](https://cutter.re/docs/contributing.html) contain the things you need to know in order to contribute code, documentation, plugins and translations to Cutter.

```

File: docs/source/contributing/code/getting-started.rst [hunk 3]
```diff
@@ -74,9 +74,8 @@ If you are experienced with such tasks, go ahead - we leave this in your hands.
 
 If you don't know how to implement something, check if any of the existing code implements similar behavior in the same widget or similar widgets. If you do copy an existing code consider why it did things the way it did, the same factors might not apply in your case or the old code was poorly written from the first place.
 
-Open a Pull request
--------------------
-
+Creating a Pull request
```
Comment (ITAYC0HEN) [inline docs/source/contributing/code/getting-started.rst:77]:
```suggestion
Create a Pull request
```

File: docs/source/contributing.rst [hunk 4]
```diff
@@ -17,6 +17,27 @@ So you like Cutter and want to get involved? Great! This part of the documentati
    <a class="btn btn-outline-success btn-lg m-2" role="button" href="contributing/plugins/getting-started.html">Plugins</a>
    <a class="btn btn-outline-info btn-lg m-2" role="button" href="contributing/translations/getting-started.html">Translations</a>
 
+steps for contributing
```
Comment (ITAYC0HEN) [inline docs/source/contributing.rst:20]:
```suggestion
Steps for Contributing
```

File: docs/source/contributing.rst [hunk 5]
```diff
@@ -17,6 +17,27 @@ So you like Cutter and want to get involved? Great! This part of the documentati
...
+======================
+- Have a good knowledge of git and github. we highly recomend to read the `GitHub-Contributing-to-a-Project <https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project>`_.
```
Comment (ITAYC0HEN) [inline docs/source/contributing.rst:22]: Please use the example I showed in the issue. Or rephrase it a little bit. This one delivers a different message

Comment (7h3qu1rkyb1t) [reply docs/source/contributing.rst:22]: ok

**Comment by 7h3qu1rkyb1t** [comment]
> 
> 
> I am not sure about having the git 101 on Cutter docs, as I prefer to link to the full tutorial. We are considering having it in its dedicated page.
> 
> I am sure that prefer it not to be in the main "cutter.re/docs/contributing" page as this page should stay clean and without practical instructions

changed to another page, to a separate one

**Comment by ITAYC0HEN** [comment]
as discussed on DM, please re-format and I'll check then :)

File: docs/source/contributing/beginners-guide.rst [hunk 6]
```diff
@@ -0,0 +1,125 @@
+********************************
+Beginners guide for contributing
+********************************
+If you want to contribute and don't know how to help this is the guide for you.
+
+
+Steps for Contributing:
+#######################
+#. Fork the project on github.
+#. clone the project to your local machine.
```
Comment (7h3qu1rkyb1t) [inline docs/source/contributing/beginners-guide.rst:10]: lines with code are getting highlighted. but couldn't solve that

**Comment by 7h3qu1rkyb1t** [comment]
> as discussed on DM, please re-format and I'll check then :)

Reformatted

**Pull request closed without merging**